src/pages/auth/redirecting.tsx

```diff
@@ -28,7 +28,8 @@ export default function Redirecting(): JSX.Element {
     } else if (!hasValidUserProfile(data)) {
       router.push('/auth/welcome?invalidProfile=true');
     } else if (data.id) {
-      router.push('/dashboard');
+      const prevPath = sessionStorage.getItem(PREV_PATH);
```

Comment:
We might need to delete the url before redirecting as the url is stale after that.